enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1147 @@
+field, as it does not provide a way to override these decisions. The new `platform.aws.vpc.subnets` field will
+continue to support automatic role selection for users who prefer a standard, opinionated default cluster subnet
+config, but only if no roles are explicitly assigned to any subnet. For IngressControllers, using automatic role
+selection enables the AWS CCM to use [AWS Subnet Discovery](#aws-subnet-discovery) as it did previously with
+`platform.aws.subnets`. For ControlPlaneLBs and ClusterNodes, the subnets are selected based on whether they are
+public or private and whether the cluster is configured as internal or external. Refer to the table below for more
+details on auto-selection.
+
+Manual role selection is when the cluster admin assigns roles to the provided subnets, and the installer
+explicitly configures the subnets to be used according to the cluster admin's role selection. Every subnet must have
+a role, and the roles labeled as "Required in Manual Mode" in the table below must be assigned at least one subnet:
+
+| Role                   | Required in Manual Mode   | Auto-Selection Component | Auto-Selected Subnets Type[^1]                             |
+|------------------------|---------------------------|--------------------------|------------------------------------------------------------|
+| ClusterNode            | Yes                       | Installer                | Private[^2]                                                |
+| EdgeNode               | No                        | Installer                | Local Zone                                                 |
+| Bootstrap              | Yes                       | Installer                | Public or Private                                          |
+| IngressControllerLB    | Yes                       | AWS CCM                  | Public for External Cluster, Private for Internal Clusters |
+| ControlPlaneExternalLB | Yes (if external cluster) | Installer (CAPA)         | Public                                                     |
+| ControlPlaneInternalLB | Yes                       | Installer (CAPA)         | Private                                                    |
+[^1]: Describes what type of subnets will be selected for this role in automatic role selection.
+[^2]: Public for OPENSHIFT_INSTALL_AWS_PUBLIC_ONLY
+
+
+### Rejecting Untagged Subnets Proposal
+
+In BYO VPCs, untagged subnets are nuisance as the AWS CCM may select them, leading to various bugs, RFEs, and
+support cases. Since this proposal introduces a new subnet install-config field, it's a great opportunity
+to introduce a new behavior which will provide a better user experience. If `platform.aws.vpc.subnets`
+is using automatic role selection (i.e. subnet discovery), the installer will inspect the VPC for untagged
+subnets not already specified in `platform.aws.vpc.subnets` and reject the installation if any exist.
```

Comment:
Our conversation around upconversion made me realize that I do not think we can reject this field only when set to `platform.aws.vpc.subnets` (and not reject with `platform.aws.subnets`). 

One thing that comes to mind is that we switch this to a warning in the installer... if users tag the "extra" subnets day 2, is that handled gracefully? 

The problem is that installer warnings get lost in the noise of the install logs and we have work to do there for it to be truly effective.

Reply:
Can we augment validation for `platform.aws.subnets`  to include this even when it is being deprecated?

Reply:
I realize now my original comment is more of an implementation detail; and also realize now how to handle it.

i havent implemented a deprecation before but the pattern i have seen is that we will rename the underlying go type to `DeprecatedSubnets` and replace all usage with the new type, including in validation code. 

Now I see we can indeed do this if we skip a validation if `DeprecatedSubnets != nil`